process/block/shardblock.go

```diff
@@ -1904,9 +1944,21 @@ func (sp *shardProcessor) createAndProcessMiniBlocksDstMe(haveTime func() bool)
 			break
 		}
 
+		hasProofForHdr := sp.proofsPool.HasProof(core.MetachainShardId, orderedMetaBlocksHashes[i])
+		shouldConsiderProofsForNotarization := sp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, orderedMetaBlocks[i].GetEpoch())
+		if !hasProofForHdr && shouldConsiderProofsForNotarization {
+			log.Trace("no proof for meta header",
+				"hash", logger.DisplayByteSlice(orderedMetaBlocksHashes[i]),
+			)
+			continue
```

Comment:
I think this should be break, continuing will iterate on the next metablock, which will break at line 1944 (the metablocks are ordered by nonce)

Reply:
missed that.. updated